gtsfm/two_view_estimator.py

```diff
@@ -160,56 +104,54 @@ def bundle_adjust(
             Optimized unit translation i2Ui1.
             Optimized verified_corr_idxs.
         """
-        i2Ti1_from_verifier: Optional[Pose3] = (
-            Pose3(i2Ri1_initial, i2Ui1_initial.point3()) if i2Ri1_initial is not None else None
-        )
-        i2Ti1_initial: Optional[Pose3] = self.__generate_initial_pose_for_bundle_adjustment(
-            i2Ti1_from_verifier, i2Ti1_prior
-        )
-
+        # Choose initial pose estimate for triangulation and BA (prior gets priority).
+        i2Ti1_initial = i2Ti1_prior.value if i2Ti1_prior is not None else None
+        if i2Ti1_initial is None and i2Ri1_initial is not None and i2Ui1_initial is not None:
+            i2Ti1_initial = Pose3(i2Ri1_initial, i2Ui1_initial.point3())
         if i2Ti1_initial is None:
             return None, None, verified_corr_idxs
 
         # Set the i1 camera pose as the global coordinate system.
         camera_class = gtsfm_types.get_camera_class_for_calibration(camera_intrinsics_i1)
-        camera_i1 = camera_class(Pose3(), camera_intrinsics_i1)
-        camera_i2 = camera_class(i2Ti1_initial.inverse(), camera_intrinsics_i2)
+        cameras = {
+            0: camera_class(Pose3(), camera_intrinsics_i1),
+            1: camera_class(i2Ti1_initial.inverse(), camera_intrinsics_i2),
+        }
 
-        # Perform data association to construct 2-view BA input.
+        # Triangulate!
+        point3d_initializer = Point3dInitializer(cameras, self._triangulation_options)
+        triangulated_indices: List[int] = []
+        triangulated_tracks: List[SfmTrack] = []
         start_time = timeit.default_timer()
-        # TODO: add flag to switch between verified and putative correspondences.
-        triangulated_tracks, triangulated_indices = self.triangulate_two_view_correspondences(
-            camera_i1=camera_i1,
-            camera_i2=camera_i2,
-            keypoints_i1=keypoints_i1,
-            keypoints_i2=keypoints_i2,
-            corr_idxs=verified_corr_idxs,
-        )
+        for j, (idx1, idx2) in enumerate(verified_corr_idxs):
```

Comment:
I have a slight preference to having this in a helper like it previously was, but upto you.

Reply:
done